include/zephyr/net/ieee802154.h

```diff
@@ -85,11 +82,15 @@ struct ieee802154_context {
 #endif
 	int16_t tx_power;
 	uint8_t sequence;
-	uint8_t ack_seq;
-	uint8_t ack_received	: 1;
-	uint8_t ack_requested	: 1;
-	uint8_t associated		: 1;
-	uint8_t _unused		: 5;
+	struct k_sem ctx_lock; /* guards all mutable context attributes unless
+				* otherwise mentioned on attribute level
+				*/
+
+	uint8_t ack_seq;	   /* guarded by ack_lock */
+	uint8_t ack_received : 1;  /* guarded by ack_lock */
+	uint8_t ack_requested : 1; /* guarded by ack_lock */
+	uint8_t _unused : 6;
+	struct k_sem ack_lock;
```

Comment:
looks like reordering the attributes this way, you mess up the structure alignment. (thus probably lose some space)

Reply:
Good point. Not the re-ordering was problematic but the introduction of the additional semaphore in the wrong place. Fixed this. There was also additional pre-existing padding in the struct (16 bytes overall) that could be removed without impacting readability. This is done in a separate commit now.